### PR TITLE
move base image to heroku16

### PR DESCRIPTION
**Pull request by pynixwang**
1. move base image to heroku16
2. do not install buildpacks, use custom buildpack for specific language not all of them.
3. install daemontools for missing setuidgid command

### CODE REVIEW

**Comment by pynixwang** [comment]
it works for my rails app, but maybe need more tests.

**Comment by pynixwang** [comment]
[Heroku-16 is now the default stack](https://devcenter.heroku.com/changelog-items/1139)

**Comment by josegonzalez** [comment]
@michaelshobbs how do you feel about this? I feel as though we should start publishing a `herokuish-14` and a `herokuish-16`?

**Comment by michaelshobbs** [comment]
I don't particularly want to maintain both nor do I think it's necessary. I'm open to arguments to the contrary however. 

cedar-14 is also based on ubuntu 14.04 which is terribly old now.

If folks absolutely must have cedar-14 then they can use the current version of herokuish and specify a `BUILDPACK_URL` if the builtin bp version is insufficient. 

Conversely, supporting multiple base image versions on a FOSS isn't really in my "top 8." 😉

Thoughts on this approach?

**Comment by josegonzalez** [comment]
We'll probably want to - at the very least - bump to a minor release of herokuish for this. Do the list of supported buildpacks need to change as well?

Can we install *just* `setguiduid` somehow, or is all of `daemontools` really required?

**Comment by rvalyi** [comment]
@josegonzalez @michaelshobbs guys sorry to bother asking you but what is your view about that statement that Dockerfile deploy is the future? Do share this view? Is it as easy to maintain a cache efficiently as Heroku buildpack do to speed up subsequent deploys?

**Comment by joshmanders** [comment]
I 👍 this.

**Comment by michaelshobbs** [comment]
In dokku, dockerfile deploys are always an option. If speed/size/finer grain control of deployment are of any concern to a particular user, I always recommend dockerfile-based deployments.

In the context of this PR however, the entire point of herokuish is to provide heroku buildpack compatibility. Thus, I think this project does in fact need to maintain that functionality.

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM heroku/cedar:14
+FROM heroku/heroku:16
 RUN curl "https://github.com/gliderlabs/herokuish/releases/download/v0.3.32/herokuish_0.3.32_linux_x86_64.tgz" \
 		--silent -L | tar -xzC /bin
-RUN /bin/herokuish buildpack install \
```
Comment (michaelshobbs) [inline Dockerfile]: Because herokuish is meant to provide heroku bp deployment compatibility, this should not be removed.

**Comment by michaelshobbs** [comment]
@josegonzalez 
> We'll probably want to - at the very least - bump to a minor release of herokuish for this. 

Agreed

> Do the list of supported buildpacks need to change as well?

I don't think so...

> Can we install just setguiduid somehow, or is all of daemontools really required?

Seems like that's how we get it via `apt`... I suppose we could compile it but it looks like a pretty small package anyways. (https://cr.yp.to/daemontools/install.html)

```
root@1117e53405d5:/# dpkg -S setuidgid
daemontools: /usr/share/man/man8/setuidgid.8.gz
daemontools: /usr/bin/setuidgid
```

**Comment by pynixwang** [comment]
gem install nio4r failed.

**Comment by pynixwang** [comment]
I switch to heroku/heroku:build-16 for development libraries and headers

**Comment by pynixwang** [comment]
runtime image has no ruby-dev package installed. add `apt-get install ruby-dev`

**Comment by michaelshobbs** [comment]
@pynixwang are you going to make these changes to the PR?

File: Dockerfile [hunk 2]
```diff
@@ -7,3 +7,4 @@ RUN /bin/herokuish buildpack install \
 	&& ln -s /bin/herokuish /exec
 COPY include/default_user.bash /tmp/default_user.bash
 RUN bash /tmp/default_user.bash && rm -f /tmp/default_user.bash
+RUN apt-get update && apt-get install -y daemontools ruby-dev
```
Comment (josegonzalez) [inline Dockerfile]: Can probably move this up and run it as the second step - but also clear out apt cache?

Comment (michaelshobbs) [reply Dockerfile]: @pynixwang todos:
- move `RUN apt-get...` above `RUN curl..`
- change the line to `RUN apt-get update -qq && apt-get install -yy -q daemontools ruby-dev`
- add `RUN apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*` to the bottom

**Comment by pynixwang** [comment]
should do some change to Dockerfile.dev?

**Comment by pynixwang** [comment]
maybe should switch to 16-build image, 16 lack development libraries and headers for build. I user it for deploy rails app, so i just test ruby, maybe lack python-dev or other package too.

**Comment by michaelshobbs** [comment]
Yes, `Dockerfile.dev` should also be updated.

**Comment by michaelshobbs** [comment]
Closing in favor of #333

**Pull request closed without merging**